### PR TITLE
Strict typescript

### PR DESCRIPTION
**Pull request by Hinton**

### CODE REVIEW

File: decisions/0014-typescript-strict.md [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+> mode family options.
+>
+> <cite>https://www.typescriptlang.org/tsconfig#strict</cite>
+
+Of particular note is the [Strict Null Checks][null]. By default TypeScript will ignore `false`,
+`null` and `undefined` allowing them to be used for any types. When `strictNullChecks` are enabled
+they must be explicitly allowed in the type definition.
+
+We are at a certain scale where enforcing more robust coding practices and standardization is more
+of a priority that it's worth raising this topic again.
```
Comment (MGibson1) [inline decisions/0014-typescript-strict.md]: Not sure what the purpose of this line is

File: decisions/0014-typescript-strict.md [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+  - desktop: 30
+  - cli: 36
+
+### Plan
+
+There are several potential pitfalls related to progressively enabling strict mode. Which will
+benefit from having a plan on how we should tackle the migration. Each category of files should have
+a proper example how how they should be migrated. And during the initial migration period we should
+not prevent people from adding new files that are not strict. What we should prevent is people
+adding new non-strict files in the category that we are currently migrating.
```
Comment (MGibson1) [inline decisions/0014-typescript-strict.md]: Not sure I agree with this, Are you worried people will not know how to adhere to strict types?

**Review by MGibson1 — CHANGES_REQUESTED** [review]
Like it in general, but a few comments

File: decisions/0014-typescript-strict.md (hunk 1 above)
Comment (Hinton) [reply decisions/0014-typescript-strict.md]: Removed.

File: decisions/0014-typescript-strict.md (hunk 2 above)
Comment (Hinton) [reply decisions/0014-typescript-strict.md]: Yes per our discussions before. It's also going to be difficult to use strict when you depend on non-strict files.

Comment (MGibson1) [reply decisions/0014-typescript-strict.md]: but I'd like to imply a best effort should be done. What do you think about 
```suggestion
a proper example how how they should be migrated. And during the initial migration period a best effort should be made to be strict-compliant, but we should
not prevent people from adding new files that are not strict. What we should prevent is people
adding new non-strict files in the category that we are currently migrating.
```

Comment (Hinton) [reply decisions/0014-typescript-strict.md]: Seems good.

**Review by MGibson1 — APPROVED** [review]

**Pull request merged**